src/vulkan/rt64_vulkan.cpp

```diff
@@ -1824,6 +1834,31 @@ namespace RT64 {
 
         VkResult res;
 
+#   ifdef RT64_BUILD_PLUGIN
```

Comment:
My suggestions here would be not to rely on compilation macros, as the RHI should not be aware at all of what type of build RT64 is.

The cleaner solution here would be to allow creating the swap chain with native handles instead as an option. This can be easily achieved by adding an optional parameter to createSwapChain that allows taking a void * to a native implementation, then reimplementing that in the concrete implementation of the Vulkan RHI class to use an external handle and skip any creation logic of the surface on its own.